### PR TITLE
Feat add fields for stats

### PR DESCRIPTION
**Pull request by Morranto**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [x] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue ##11081

## What this PR does / why we need it:

### CODE REVIEW

File: pkg/util/trace/impl/motrace/statistic/stats_array.go [hunk 1]
```diff
@@ -215,53 +215,51 @@ type statsInfoKey struct{}
...
+	//S3WriteBytes            uint
 
-	//temporary fields which is used to compute data above
-
-	planStart time.Time
-
-	compileStart time.Time
+	ParseStartTime   time.Time `json:"ParseStartTime"`
+	PlanStartTime    time.Time `json:"PlanStartTime"`
+	CompileStartTime time.Time `json:"CompileStartTime"`
```
Comment (ouyuanning) [inline pkg/util/trace/impl/motrace/statistic/stats_array.go:231]: Run的Start和End也加一下吧

**Review by xzxiong — COMMENTED** [review]
需要一个 benchmark 基本说明一下 statistic.StatsInfo 的消耗，摸摸底。

File: pkg/util/trace/impl/motrace/statistic/stats_array_test.go [hunk 2]
```diff
@@ -325,3 +327,16 @@ func TestStatsArray_InitIfEmpty(t *testing.T) {
 		})
 	}
 }
+
+func BenchmarkStatsInfo(b *testing.B) {
```
Comment (xzxiong) [inline pkg/util/trace/impl/motrace/statistic/stats_array_test.go:331]: Benchmark 不是这么写的

Comment (xzxiong) [reply pkg/util/trace/impl/motrace/statistic/stats_array_test.go:331]: example
```
func BenchmarkTestFunc(b *testing.B) {
    // prepare your job
    // ...

    // start your benchmakr
    b.ResetTimer()
    for i := 0; i < b.N; i++ {
        your_target_func()
    }
}
```

Comment (Morranto) [reply pkg/util/trace/impl/motrace/statistic/stats_array_test.go:331]: ok，understand

Comment (Morranto) [reply pkg/util/trace/impl/motrace/statistic/stats_array_test.go:331]: 已经修改

**Review by ouyuanning — APPROVED** [review]

**Comment by Morranto** [comment]
企业微信截图_9c6a4901-fc37-408a-917b-78a383e18ac6

**Review by xzxiong — APPROVED** [review]

**Review by aunjgr — APPROVED** [review]

**Review by nnsgmsone — APPROVED** [review]

**Review by qingxinhome — APPROVED** [review]

**Pull request merged**